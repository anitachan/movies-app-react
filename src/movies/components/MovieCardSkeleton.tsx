import { Box, Skeleton } from '@mui/material';

export const MovieCardSkeleton = () => (
	<Box sx={{ position: 'relative' }}>
		<Skeleton variant="rectangular" width="100%" height={200} sx={{ borderRadius: 4 }} />
		<Box
			sx={{
				bgcolor: 'rgba(0, 0, 0, 0.7)',
				color: 'white',
				p: 2,
				mt: '-4px',
				display: 'flex',
				flexDirection: 'row',
				justifyContent: 'space-between',
				alignItems: 'center',
				borderBottomLeftRadius: 4,
				borderBottomRightRadius: 4,
			}}
		>
			<Box sx={{ flex: 1 }}>
				<Skeleton variant="text" width="80%" />
				<Skeleton variant="text" width="100%" />
				<Skeleton variant="text" width="40%" />
			</Box>
			<Skeleton variant="rectangular" width={80} height={30} sx={{ borderRadius: 1, ml: 2 }} />
		</Box>
	</Box>
);
