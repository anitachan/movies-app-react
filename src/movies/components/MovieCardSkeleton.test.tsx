import { render } from '@testing-library/react';
import { MovieCardSkeleton } from './MovieCardSkeleton';

describe('MovieCardSkeleton', () => {
	test('MovieCardSkeleton matches snapshot', () => {
		const { container } = render(<MovieCardSkeleton />);
		expect(container).toMatchSnapshot();
	});
});
