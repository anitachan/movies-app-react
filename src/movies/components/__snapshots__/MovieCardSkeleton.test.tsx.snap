// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardSkeleton MovieCardSkeleton matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-79elbk"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-h33yhd-MuiSkeleton-root"
      style="width: 100%; height: 200px;"
    />
    <div
      class="MuiBox-root css-ism2yd"
    >
      <div
        class="MuiBox-root css-1rr4qq7"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
          style="width: 80%;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
          style="width: 100%;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
          style="width: 40%;"
        />
      </div>
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1tzq9ic-MuiSkeleton-root"
        style="width: 80px; height: 30px;"
      />
    </div>
  </div>
</div>
`;
